ASP.NET API/WAVC_WebApi/Controllers/FriendRequestsController.cs

```diff
@@ -1,134 +1,83 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Authorization;
-using Microsoft.AspNetCore.Http;
+﻿using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
 using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using WAVC_WebApi.Data;
+using WAVC_WebApi.Hubs;
+using WAVC_WebApi.Hubs.Interfaces;
 using WAVC_WebApi.Models;
-using WAVC_WebApi.Models.MiscellaneousModels;
+using WAVC_WebApi.Models.HelperModels;
 
 namespace WAVC_WebApi.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
+    [Authorize]
     public class FriendRequestsController : ControllerBase
     {
-        private readonly ApplicationDbContext _context;
-        private UserManager<ApplicationUser> _userManager;
+        private readonly ApplicationDbContext _dbContext;
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IHubContext<FriendRequestHub, IFriendRequestClient> _friendRequestHubContext;
 
-        public FriendRequestsController(ApplicationDbContext context, UserManager<ApplicationUser> userManager)
+        public FriendRequestsController(ApplicationDbContext context, UserManager<ApplicationUser> userManager, IHubContext<FriendRequestHub, IFriendRequestClient> hubContext)
         {
-            _context = context;
+            _dbContext = context;
             _userManager = userManager;
+            _friendRequestHubContext = hubContext;
         }
 
 
         // GET: api/FriendRequests
         [HttpGet]
-        public async Task<ActionResult<IEnumerable<FriendRequest>>> GetFriendRequests()
+        public async Task<ActionResult<IEnumerable<Object>>> GetFriendRequests()
         {
-            return await _context.FriendRequests.ToListAsync();
-        }
+            var user = await _userManager.FindByIdAsync(User.Identity.Name);
```

Comment:
It won't work until I change the authorization method. Refactor of one line will be easy.

Reply:
Are you 100% sure it won't work, or you just geussing it won't? (I have a feeling that it might do). If it won't work, then it sucks that one changes will break other parts of code.

Reply:
It won't work because when you're using jwt authorization server doesn't store information about user sessions. You have to configure user claims explicitly. Maybe there is a way to hack it but there's no time for that.

Reply:
Where did you get that session from? I told you before, ASP.NET doesn't use session!

Reply:
I'm just gonna test it.